### PR TITLE
Multiple analytics modules: allow pub-defined event filters; do not block auction for analytics

### PR DESCRIPTION
**Pull request by dgirardi**
## Type of change

- [x] Feature

## Description of change

This replaces the hardcoded list of events that are forwarded to analytics adapters with a whitelist and/or blacklist provided by the publisher:

```
pbjs.enableAnalytics([{
   provider: 'rubicon',
   includeEvents: [
       'bidWon', 
       'auctionInit', 
       // ...
   ],
   excludeEvents: [
       'auctionDebug', 
       // ...
   ],
   options: {
     /...
   }
}])
```

Additionally, event handlers for analytics are deferred until some time has passed since the last event, to avoid blocking the auction (see https://github.com/prebid/Prebid.js/issues/9110). The debounce delay is fixed at 100ms; I am not sure if there is a use case to make it configurable.

## Other information

Resolves https://github.com/prebid/Prebid.js/issues/9014
Resolves https://github.com/prebid/Prebid.js/issues/9110
Documentation: https://github.com/prebid/prebid.github.io/pull/4082

### CODE REVIEW

**Review by ncolletti — APPROVED** [review]
these changes look good to me, will wait for a second review before merge

**Review by patmmccann — APPROVED** [review]

**Pull request merged**
